src/saveload/afterload.cpp

```diff
@@ -3196,9 +3196,9 @@ bool AfterLoadGame()
 			}
 		}
 
-		/* Refresh all level crossings to bar adjacent crossing tiles. */
+		/* Refresh all level crossings to bar adjacent crossing tiles, if needed. */
 		for (TileIndex tile = 0; tile < Map::Size(); tile++) {
-			if (IsLevelCrossingTile(tile)) UpdateLevelCrossing(tile, false, true);
+			if (IsLevelCrossingTile(tile)) UpdateLevelCrossing(tile, false, false);
```

Comment:
```suggestion
			if (IsLevelCrossingTile(tile)) UpdateLevelCrossing(tile, false);
```
And I think you can also remove the conversion in 1988-1993.